test/unwrap.jl

```diff
@@ -0,0 +1,140 @@
+        @test (A_wrapped + d) ≈ A_unwrapped
+
+        v_unwrapped_range = collect(linspace(0, 4, 7))
+        A_unwrapped_range = v_unwrapped_range .+ v_unwrapped_range'
+        test_range = convert(T, 2)
+        A_wrapped_range = A_unwrapped_range .% test_range
+
+        test_unwrapped_range = unwrap(A_wrapped_range, dims=1:2; range=test_range)
+        d = first(A_unwrapped_range) - first(test_unwrapped_range)
+        @test (test_unwrapped_range + d) ≈ A_unwrapped_range
+
+        # Test wrap_around
+        # after unwrapping, pixels at borders should be equal to corresponding pixels
+        # on other side
+        wrap_around = (true, true)
+        wa_vec = linspace(0, 4convert(T, π), 10)
+        wa_uw = wa_vec .+ zeros(10)'
+        # make periodic
+        wa_uw[end, :] = wa_uw[1, :]
+        wa_w = wa_uw .% (2π)
+        wa_test = unwrap(wa_w, dims=1:2, wrap_around=wrap_around, seed=0)
+        # with wrap-around, the borders should be equal, but for this problem the
+        # image may not be recovered exactly
+        @test wa_test[:, 1] ≈ wa_test[:, end]
+        @test wa_test[end, :] ≈ wa_test[1, :]
+        # In this case, calling unwrap w/o wrap_around does not recover the borders
+        wa_test_nowa = unwrap(wa_w, dims=1:2)
+        @test !(wa_test_nowa[end, :] ≈ wa_test_nowa[1, :])
+
+    end
+end
+
+@testset "Unwrap 3D" begin
+    types = (Float32, Float64, BigFloat)
+    f(x, y, z) = 0.1x^2 - 2y + 2z
+    f_wraparound2(x, y, z) = 5*sin(x) + 2*cos(y) + z
+    f_wraparound3(x, y, z) = 5*sin(x) + 2*cos(y) - 4*cos(z)
+    for T in types
+        grid = linspace(zero(T), 2convert(T, π), 11)
+        f_uw = f.(grid, grid', reshape(grid, 1, 1, :))
+        f_wr = f_uw .% (2convert(T, π))
+        uw_test = unwrap(f_wr, dims=1:3)
+        offset = first(f_uw) - first(uw_test)
+        @test (uw_test+offset) ≈ f_uw rtol=eps(T) #oop, nowrap
+        # test in-place version
+        unwrap!(f_wr, dims=1:3)
+        offset = first(f_uw) - first(f_wr)
+        @test (f_wr+offset) ≈ f_uw rtol=eps(T) #ip, nowrap
+
+        f_uw = f_wraparound2.(grid, grid', reshape(grid, 1, 1, :))
+        f_wr = f_uw .% (2convert(T, π))
+        uw_test = unwrap(f_wr, dims=1:3)
+        offset = first(f_uw) - first(uw_test)
+        @test (uw_test+offset) ≈ f_uw #oop, 2wrap
+        # test in-place version
+        unwrap!(f_wr, dims=1:3, wrap_around=(true, true, false))
+        offset = first(f_uw) - first(f_wr)
+        @test (f_wr+offset) ≈ f_uw #ip, 2wrap
+
+        f_uw = f_wraparound3.(grid, grid', reshape(grid, 1, 1, :))
+        f_wr = f_uw .% (2convert(T, π))
+        uw_test = unwrap(f_wr, dims=1:3, wrap_around=(true, true, true))
+        offset = first(f_uw) - first(uw_test)
+        @test (uw_test+offset) ≈ f_uw #oop, 3wrap
+        # test in-place version
+        unwrap!(f_wr, dims=1:3, wrap_around=(true, true, true))
+        offset = first(f_uw) - first(f_wr)
+        @test (f_wr+offset) ≈ f_uw #oop, 3wrap
+    end
+end
```

Comment:
IIUC, `calculate_pixel_reliability` is special-cased for 2d and 3d. So maybe even test for even more dimensions?